### PR TITLE
data structure and implementation

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
In this PR I added the structure for the data module:

- Moved classes from "app" to "data" module, with some refactor done
- Removed retrofit from app gradle and added to data gradle.
- Implementations for the repository and service interfaces in "domain".

Data module structure:

data module

### CODE REVIEW

**Review by lucianoeli — APPROVED** [review]

File: domain/src/main/java/com/globant/domain/usecase/implementation/GetAllCharactersUseCaseImpl.kt [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+import com.globant.domain.service.CharactersService
 import com.globant.domain.usecase.GetAllCharactersUseCase
+import com.globant.domain.util.Constants
 import com.globant.domain.util.Result
 
-class GetAllCharactersUseCaseImpl(private val characterRepository: CharacterRepository): GetAllCharactersUseCase {
-    override fun invoke(): Result<List<Character>> = characterRepository.getAllCharacters()
+class GetAllCharactersUseCaseImpl(private val characterService: CharactersService) : GetAllCharactersUseCase {
+    override fun invoke(): Result<List<Character>> {
+        characterService.getAllCharacters().let {
```
Comment (SartiMau) [inline domain/src/main/java/com/globant/domain/usecase/implementation/GetAllCharactersUseCaseImpl.kt]: the when must be in the serviceImpl

here you will just do something like

override fun invoke(): Result<List<Character>> = characterService.getAllCharacters()

File: domain/src/main/java/com/globant/domain/usecase/implementation/GetSpecificCharacterUseCaseImpl.kt [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+import com.globant.domain.service.CharactersService
 import com.globant.domain.usecase.GetSpecificCharacterUseCase
+import com.globant.domain.util.Constants.NOT_FOUND
 import com.globant.domain.util.Result
 
-class GetSpecificCharacterUseCaseImpl(private val characterRepository: CharacterRepository): GetSpecificCharacterUseCase {
-    override fun invoke(characterId: String): Result<List<Character>> = characterRepository.getSpecificCharacter(characterId)
+class GetSpecificCharacterUseCaseImpl(private val characterService: CharactersService) : GetSpecificCharacterUseCase {
+    override fun invoke(characterId: String): Result<List<Character>> {
+        characterService.getSpecificCharacter(characterId).let {
```
Comment (SartiMau) [inline domain/src/main/java/com/globant/domain/usecase/implementation/GetSpecificCharacterUseCaseImpl.kt]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: domain/src/main/java/com/globant/domain/usecase/implementation/GetAllCharactersUseCaseImpl.kt (hunk 1 above)
Comment (MarcosCGlobant) [reply domain/src/main/java/com/globant/domain/usecase/implementation/GetAllCharactersUseCaseImpl.kt]: refactored, thanks

File: domain/src/main/java/com/globant/domain/usecase/implementation/GetSpecificCharacterUseCaseImpl.kt (hunk 2 above)
Comment (MarcosCGlobant) [reply domain/src/main/java/com/globant/domain/usecase/implementation/GetSpecificCharacterUseCaseImpl.kt]: refactored, thanks

**Review by SartiMau — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**